src/unstract/sdk/index.py

```diff
@@ -293,82 +293,101 @@ def index(
             if not extracted_text:
                 raise IndexingError("No text available to index")
 
-            full_text = [
-                {
-                    "section": "full",
-                    "text_contents": extracted_text,
-                }
-            ]
-
-            # Check if chunking is required
-            documents = []
-            for item in full_text:
-                text = item["text_contents"]
-                self.tool.stream_log("Indexing file...")
-                document = Document(
-                    text=text,
-                    doc_id=doc_id,
-                    metadata={"section": item["section"]},
-                )
-                document.id_ = doc_id
-                documents.append(document)
-            self.tool.stream_log(f"Number of documents: {len(documents)}")
-
-            if doc_id_found:
-                # Delete the nodes for the doc_id
-                try:
-                    vector_db.delete(ref_doc_id=doc_id)
-                    self.tool.stream_log(f"Deleted nodes for {doc_id}")
-                except Exception as e:
-                    self.tool.stream_log(
-                        f"Error deleting nodes for {doc_id}: {e}",
-                        level=LogLevel.ERROR,
-                    )
-                    raise SdkError(f"Error deleting nodes for {doc_id}: {e}") from e
+            self.index_to_vector_db(
+                vector_db=vector_db,
+                embedding=embedding,
+                chunk_size=chunk_size,
+                chunk_overlap=chunk_overlap,
+                doc_id=doc_id,
+                text_to_idx=extracted_text,
+                doc_id_found=doc_id_found,
+            )
+            return doc_id
+        finally:
+            vector_db.close()
```

Comment:
@chandrasekharan-zipstack if we return in line 305 we will never even reach this line. How will the vector DB be closed in such case? Shouldn't this be included inisde try as well?

Also there isn't even a catch block. How will this behave in case of error?

Reply:
@ritwik-g the `finally` block purely takes care of closing the vector DB connection in case of any error. The reason its added here is because [this](https://github.com/Zipstack/unstract-sdk/blob/4ee2c3f46c453fdaf404d2c899e01dc15486a3e3/src/unstract/sdk/index.py#L244-L248) is where we create the connection as well. In spite of the return, I believe the logic in the `finally` block gets executed

Reply:
Yes finally block executes the last both in case of exceptions and normal return flows.